### ISSUE TITLE
TypeScript 3.6.2 complains about uninitialized enum values

### ISSUE THREAD

**Issue by ShawnTalbert** [headintheclouddev/typings-suitescript-2.0#117]
latest TSC **3.6.2** `N/record.d.ts`  fails compilation because the `Type` enumeration has members lacking assigned values:

    node_modules/@hitc/netsuite-types/N/record.d.ts:873:5 - error TS1061: Enum member must have initializer.

TS 3.5.1 does not complain, so this must be a new restriction?

**Comment by johnogle222** [comment]
It always felt weird to me that some of these were missing their string values anyways. This (and the other enums in the project) should probably be changed to an `Interface` with a bunch of `string` values as it isn't a full ActionScript enum. For example:
```typescript
export interface Type {
  ACCOUNT: string;
  ACCOUNTING_BOOK: string;
  ...
}
```

Thoughts?

**Comment by ShawnTalbert** [comment]
My short comment is - can't/should we just drop ALL enumeration value **assignments** in these **declaration** files?

The more long winded version:
Similarly, I found it kinda weird that any values are defined there at all - I figured capturing the abstraction (enum **declaration** not **definition**) is what should be in the type declarations. We don't control the implementation on the NS side (here the _implementation_ heppans to be just string values but the concept holds, yes?). By defining values it assumes implementation details that the NS documentation doesn't express and NS could change at any time.

For example, `Type.ACCOUNT` may indeed resolve to the string `'account'` behind the scenes but  we're only supposed to code against the abstraction `Type.ACCOUNT`. As far as I can tell from the docs, NS is trying to encourage programming to the enumeration abstractions exclusively, and slowly forget that they may (or may not in the future!) resolve to strings from SS1. 

What would you think of getting rid of the values entirely (optionally leaving them as inline comments or something), so we're not _assuming_ NS implementation?

Technically, think NS considers `record.Type` to be a JS object as opposed to an actual TypeScript `enumeration` so in that regard using `interface` may be more accurate. However, we should think carefully about how this 'enum' is actually used, so that the TS type system helps rather than hinders. For example `record.load( { type: xxx })` would need to have the `type` parameter changed - currently it accepts an enumeration type which tends to work well, no? Long ago, switching this to an interface would have been a big pain in the butt for users. However today I suppose we could use `keyof Type` to type that parameter?

**Comment by johnogle222** [comment]
Yeah that all sounds fair to me. So we would end up with something like:
```typescript
declare enum Type {
  ACCOUNT, //account
  ACCOUNTING_BOOK, //accounting_book
  ...
  DRIVERS_LICENSE,
}
export Type
```

Do you know if that would squelch the warning?

**Comment by ShawnTalbert** [comment]
I think this is resolved in the latest release. 

p.s. it would be nice if you tagged the releases so that the commit reflected by the npm release is traceable to here in github?

**Comment by johnogle222** [comment]
Yeah just by looking through the commit history it should be fixed in 2019.2.2

@MrRob you can see the release tagging documentation here: https://help.github.com/en/articles/creating-releases
You should be able to skip steps 7 and 8 as we don't publish any binaries.

**Comment by MrRob** [comment]
@johnogle222 Oh cool, I'll try that next time :)
